File: solutions/cpp/secret-handshake/2/secret_handshake.cpp
#include "secret_handshake.h"
#include <vector>
#include <array>
#include <string>
#include <cstddef>
#include <bitset>
#include <algorithm>

using namespace std;

namespace secret_handshake {

vector<string> commands(int code){
    bitset<5> cmds_b(code);
    array<string, 4> words{"wink", "double blink", 
        "close your eyes", "jump"};
    vector<string> actions;
    for(size_t i=0; i<cmds_b.size()-1; ++i){
        if (cmds_b.test(i))
            actions.push_back(words.at(i));
    }
    if (cmds_b.test(4))
        reverse(actions.begin(), actions.end());
    return actions;
  }
}  // namespace secret_handshake

File: solutions/julia/secret-handshake/3/secret-handshake.jl
function secret_handshake(code)
    actions = ["wink", "double blink", "close your eyes", "jump"]
    bits = reverse(bitstring(UInt16(code)))
    output = [actions[i] for i in 1:4 if bits[i]=='1']
    bits[5] == '1' && return reverse(output)
    output                                
end
